### ISSUE TITLE
changing scalar enum type value caused assertion error when migrating

### ISSUE THREAD

**Issue by tofusoul** [edgedb/edgedb#5059]
- EdgeDB Version: 2.10+4a92d32
- EdgeDB CLI Version: 2.2.6+b869667
- OS Version: Fedora Linux

Steps to Reproduce:

1. changed enum in default.esdl  scalar type Unit extending enum<kg, g, L, ml, tsp, tbls, cup,> to enum<kg, g, L, ml, tsp, tbsp, cup,>
2. edgedb migration create
3. edgedb migrate

migration file: 
{
  ALTER SCALAR TYPE default::Unit EXTENDING enum<kg, g, L, ml, tsp, tbsp, cup>;
}

```
Schema:
 Server traceback:
      Traceback (most recent call last):
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/server/compiler_pool/worker_proc.py", line 55, in worker
          res = meth(*args)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/server/compiler_pool/worker.py", line 185, in compile_in_tx
          units, cstate = COMPILER.compile_in_tx(cstate, *args, **kwargs)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 2338, in compile_in_tx
          return self._compile(ctx=ctx, source=source), ctx.state
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 1812, in _compile
          return self._try_compile(ctx=ctx, source=source)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 1876, in _try_compile
          comp, capabilities = self._compile_dispatch_ql(
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 1734, in _compile_dispatch_ql
          query = self._compile_ql_migration(
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 1060, in _compile_ql_migration
          query = self._compile_and_apply_ddl_stmt(ctx, ql)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 956, in _compile_and_apply_ddl_stmt
          block, new_types, config_ops = self._process_delta(ctx, delta)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 390, in _process_delta
          schema = pgdelta.apply(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 6603, in apply
          schema = super().apply(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 1510, in apply
          schema = self.apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 148, in apply_subcommands
          schema = super().apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 1537, in apply_subcommands
          schema = objop.apply(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 2996, in apply
          schema = self._create_innards(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 2964, in _create_innards
          return self.apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 148, in apply_subcommands
          schema = super().apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 946, in apply_subcommands
          schema = op.apply(schema, context=context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 3463, in apply
          schema = self._alter_begin(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 2581, in _alter_begin
          schema = self._undo_everything(schema, context, objs, props)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 2468, in _undo_everything
          schema = acmd.apply(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 1139, in apply
          schema = self.apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 148, in apply_subcommands
          schema = super().apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 946, in apply_subcommands
          schema = op.apply(schema, context=context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 3341, in apply
          schema = super().apply(schema, context=context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 3464, in apply
          schema = self._alter_innards(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 3093, in _alter_innards
          return self.apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 148, in apply_subcommands
          schema = super().apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 946, in apply_subcommands
          schema = op.apply(schema, context=context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 3464, in apply
          schema = self._alter_innards(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 5251, in _alter_innards
          schema = super()._alter_innards(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 3093, in _alter_innards
          return self.apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 148, in apply_subcommands
          schema = super().apply_subcommands(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 946, in apply_subcommands
          schema = op.apply(schema, context=context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/schema/delta.py", line 3146, in apply
          schema = self._alter_begin(schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 5164, in _alter_begin
          self._alter_pointer_type(self.scls, schema, orig_schema, context)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 4059, in _alter_pointer_type
          self.create_inhview(schema, context, source, alter_ancestors=False)
        File "/home/tofu/.local/share/edgedb/portable/2.10/lib/python3.10/site-packages/edb/pgsql/delta.py", line 3001, in create_inhview
          assert has_table(obj, schema)
      AssertionError
```

**Comment by msullivan** [comment]
This doesn't quite have enough information for me to reproduce (since it lacks the initial schema), but my guess is that this is going to be fixed by #5457.

**Comment by msullivan** [comment]
OK, yeah, I managed to repro it (I think) and it is fixed by that. Reopen if you think I'm mistaken
